### PR TITLE
[-] Performance - Improve the speed of listing the records by executing their count into another request

### PR DESCRIPTION
**Pull request by root-io**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request
- [x] Write changes made in the CHANGELOG.md
- [ ] Create automatic tests
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by root-io** [comment]
[(13) Users want to dissociate count (meta) from data when listing all records](https://trello.com/c/Nms1rxL2/2666-13-users-want-to-dissociate-count-meta-from-data-when-listing-all-records)

File: services/resources-getter.js [hunk 1]
```diff
@@ -138,7 +133,6 @@ function ResourcesGetter(model, opts, params) {
           _.each(schema.fields, function (field) {
             if (field.search) {
               try {
-                field.search(countOpts, params.search);
```
[the remark this answers is not in the document]
Comment (arnaudbesnier) [reply services/resources-getter.js]: @root-io ?

File: services/resources-getter.js [hunk 2]
```diff
@@ -150,14 +144,44 @@ function ResourcesGetter(model, opts, params) {
...
+    var include = queryBuilder.getIncludes(model, fieldNamesRequested);
+
+    return getWhere()
+      .then(function (where) {
+        var countOpts = {
+          include: include,
+          where: where
+        };
+
+        if (!fieldNamesRequested[0]) {
```
Comment (arnaudbesnier) [inline services/resources-getter.js]: This criteria is error prone IMO: `fieldNamesRequested` is not always available information (in Smart Views for example). Can't you check that the schema as a primary key? I'll try this.

File: services/resources-getter.js [hunk 3]
```diff
@@ -150,14 +144,26 @@ function ResourcesGetter(model, opts, params) {
...
+    var include = queryBuilder.getIncludes(model, fieldNamesRequested);
+
+    return getWhere()
+      .then(function (where) {
+        var countOpts = {
+          include: include,
+          where: where
+        };
+
+        return scope.count(countOpts);
```
[the remark this answers is not in the document]
Comment (root-io) [reply services/resources-getter.js]: Fixed

File: services/resources-getter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (root-io) [reply services/resources-getter.js]: Added, thanks 👍

**Pull request merged**